functions/go/ensure-name-segment/ensure_name_segment.go

```diff
@@ -0,0 +1,203 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/kustomize/api/filters/prefixsuffix"
+	"sigs.k8s.io/kustomize/api/resid"
+	"sigs.k8s.io/kustomize/api/resmap"
+	"sigs.k8s.io/kustomize/api/resource"
+	"sigs.k8s.io/kustomize/api/types"
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+	k8syaml "sigs.k8s.io/yaml"
+)
+
+const (
+	fnConfigGroup      = "fn.kpt.dev"
+	fnConfigVersion    = "v1alpha1"
+	fnConfigAPIVersion = fnConfigGroup + "/" + fnConfigVersion
+	fnConfigKind       = "EnsureNameSegment"
+)
+
+type EnsureNameSegment struct {
+	yaml.ResourceMeta `json:",inline" yaml:",inline"`
+	// Segment is the desired name segment.
+	Segment string `json:"segment" yaml:"segment"`
+	// ActionOnNotFound controls the desired action when the desired segment is not found in the name.
+	// If not specified, prepend will be the default.
+	ActionOnNotFound ActionMode `json:"actionOnNotFound,omitempty" yaml:"actionOnNotFound,omitempty"`
+
+	FieldSpecs []types.FieldSpec `json:"fieldSpecs,omitempty" yaml:"fieldSpecs,omitempty"`
```

Comment:
I am guessing for most common use cases name in the metadata of a KRM resource, users won't have to specify this. Right ?

Reply:
Right, for common use cases, users don't need to specify it and the default ones should do the job for them.
I imagine users will use ConfigMap for common use cases.